inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/utils/BufferQueue.java

```diff
@@ -61,6 +61,19 @@ public A pollRecord() {
         return record;
     }
 
+    /**
+     * Take record
+     */
+    public A takeRecord() {
+        try {
+            A record = queue.take();
```

Comment:
take operation will lock current thread.
Please confirm this feature.

Reply:
It's the take() method specially used, mainly to block until there is data in the queue before proceeding with the next processing